app/server/appsmith-server/src/test/java/com/appsmith/server/services/UserServiceTest.java

```diff
@@ -415,12 +415,36 @@ public void createUserWithInvalidEmailAddress() {
     @WithUserDetails(value = "api_user")
     public void updateNameOfUser() {
         UserUpdateDTO updateUser = new UserUpdateDTO();
-        updateUser.setName("New name of api_user");
+        updateUser.setName("New name of api user");
         StepVerifier.create(userService.updateCurrentUser(updateUser, null))
                 .assertNext(user -> {
                     assertNotNull(user);
                     assertThat(user.getEmail()).isEqualTo("api_user");
-                    assertThat(user.getName()).isEqualTo("New name of api_user");
+                    assertThat(user.getName()).isEqualTo("New name of api user");
+                })
+                .verifyComplete();
+    }
+
+    @Test
+    @WithUserDetails(value = "api_user")
+    public void updateNameOfUser_WithNotAllowedSpecialCharacter_InvalidName() {
+        UserUpdateDTO updateUser = new UserUpdateDTO();
+        updateUser.setName("invalid name@symbol");
+        StepVerifier.create(userService.updateCurrentUser(updateUser, null))
+                .expectErrorMatches(throwable -> throwable instanceof AppsmithException && throwable.getMessage()
+                .contains(AppsmithError.INVALID_PARAMETER.getMessage(FieldName.NAME)))
+                .verify();
```

Comment:
This formatting was hard to follow. The way the `.extectErrorMatches`, `.contains` and `.verify` were lined up, I immediately though that `.verify` is called on the return value of `.contains`, and `.contains` is called on return value of `.expectErrorMatches`, which isn't the case.

Let's fix the indentation to express the intent better, and make reading the code easier. 🎉

```suggestion
        StepVerifier.create(userService.updateCurrentUser(updateUser, null))
            .expectErrorMatches(throwable ->
                throwable instanceof AppsmithException
                    && throwable.getMessage().contains(AppsmithError.INVALID_PARAMETER.getMessage(FieldName.NAME))
            )
            .verify();
```

What say?